### DISCUSSION TITLE
Communication timeout during homing X/Y

### DISCUSSION THREAD

**Discussion by WatWall** [xblax/flashforge_ad5m_klipper_mod#133] [Q&A]
I get this error "Communication timeout during homing X" when I start printing. 
Sometimes it happens on Y axis too. I retry printing 4-5 times and it works
Is there any fix for this?

```
File opened:Prints/flowrate_0_PLA_29m56s.gcode Size:451562
00:42:21 
File selected
00:42:24 
!! Communication timeout during homing x
00:42:24 
!! Communication timeout during homing x
00:42:24 
!! Communication timeout during homing x
00:42:24 
echo: "extruder" not hot enough
00:42:24 
echo: Printer not homed
00:42:24 
echo: "extruder" not hot enough
00:42:24 
!! Must home axis first: 106.462 74.901 218.764 [0.000]
00:42:24 
!! Must home axis first: 106.462 74.901 218.764 [0.000]
00:42:24 
!! Must home axis first: 106.462 74.901 218.764 [0.000]
00:42:24 
!! Must home axis first: 106.462 74.901 218.764 [0.000]
00:42:29 
```

**Comment by KaruroChori** [comment]
Are you using the version with klipperscreen?

**Reply by WatWall** [reply]
Yes, Installed the latest with klipperscreen

**Reply by KaruroChori** [reply]
That error is due to latency in the linux side of klipper while trying to keep up with the firmware.  
Klipperscreen might be a bit too much for the low spec board running it, and this jitter in latency is likely to cause the issue you are seeing from time to time.

If you can work over a network and you want a more stable environment I would suggest to stick with the headless version and control it remotely.   

@consp  made some progress recently in porting over a more lightweight UI; hopefully that should solve this problem in some future release.

**Reply by WatWall** [reply]
Thanks. I installed the lite version. I just got the error once when I first opened the printer and homed manually. Then I did 4 test prints in a row and they all started without any error. It looks better now.
